pkg/controller/clusterautoscaler/clusterautoscaler_controller.go

```diff
@@ -149,6 +151,15 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	// generated for these cluster scoped objects out of the default namespace.
 	caRef := r.objectReference(ca)
 
+	// Validate the ClusterAutoscaler early and requeue if any errors are found.
```

Comment:
Can `r.validate` ever be `nil` at this point?

Reply:
Only if you didn't use `NewReconciler()` and purposefully constructed your own reconciler without a validator. The same goes for all the fields though, e.g. the manager, client, etc.